ai-ml/jupyterhub/addons.tf

```diff
@@ -0,0 +1,244 @@
+module "eks_blueprints_kubernetes_addons" {
+  source = "github.com/aws-ia/terraform-aws-eks-blueprints-addons?ref=90a70ba"
+
+  cluster_name      = module.eks.cluster_name
+  cluster_endpoint  = module.eks.cluster_endpoint
+  cluster_version   = module.eks.cluster_version
+  oidc_provider_arn = module.eks.oidc_provider_arn
+  #---------------------------------------
+  # Amazon EKS Managed Add-ons
+  #---------------------------------------
+  eks_addons = {
+    aws-ebs-csi-driver = {
+      service_account_role_arn = module.ebs_csi_driver_irsa.iam_role_arn
+      preserve                 = true
+      most_recent              = true
+    }
+    coredns = {
+      most_recent = true
+      preserve    = true
+    }
+    kube-proxy = {
+      most_recent = true
+      preserve    = true
+    }
+  }
+  
+  #---------------------------------------
+  # Karpenter Autoscaler for EKS Cluster
+  #---------------------------------------
+  enable_karpenter                  = true
+  karpenter_enable_spot_termination = true
+  karpenter_node = {
+    create_iam_role          = true
+    iam_role_use_name_prefix = false
+    # We are defining role name so that we can add this to aws-auth during EKS Cluster creation
+    iam_role_name = local.karpenter_iam_role_name
```

Comment:
remove, you can get name as output, I will show below

Reply:
done